### PR TITLE
Generate scenarios for isolation testing.

### PR DESCRIPTION
**Pull request by brysonjbest**
- [x] Add multiselect handling to klamm inputs.
- [x] Add new 'Isolation Tests' tab for csv creation
- [x] Users can now generate a csv with up to 1000 possible scenarios, based on the inputs pulled from klamm.
- [x] Users can define specific fields they wish to remain unchanged for these tests. All other fields are generated from the constraints of inputs pulled from klamm.
- [x] Generated tests are created with descriptive filenames indicating the values that have been hard-defined by the user.

See https://github.com/bcgov/brms-api/pull/38

### CODE REVIEW

File: app/components/InputStyler/InputStyler.tsx [hunk 1]
```diff
@@ -207,6 +217,12 @@ export default function InputStyler(
         );
     }
   } else {
+    const allObjects = Array.isArray(value) && value.every((item) => typeof item === "object" && item !== null);
+    if (Array.isArray(value) && !allObjects) {
+      const stringValue = value.filter((item) => typeof item !== "object" || item === null).join(", ");
+      return <ReadOnlyStringDisplay show value={stringValue} />;
+    }
```
Comment (timwekkenbc) [inline app/components/InputStyler/InputStyler.tsx:226]: Can you comment up what this is doing? I like to add comments where things aren't immediately obvious (although I'm sure I miss lots - "do as I say, not as I do" lol).

File: app/utils/api.ts [hunk 2]
```diff
@@ -401,3 +402,49 @@ export const getBREFieldFromName = async (fieldName: string): Promise<KlammBREFi
...
+    let filename = "";
+
+    const scenarioName = simulationContext ? `${generateDescriptiveName(simulationContext)}` : "";
+    const MAX_FILENAME_LENGTH = 250;
+
+    if (`${scenarioName + filepath}`.length > MAX_FILENAME_LENGTH && simulationContext) {
+      filename = `${(generateDescriptiveName(simulationContext, 5) + filepath).replace(/\.json$/, ".csv")}`;
+    } else {
+      filename = `${(scenarioName + filepath).replace(/\.json$/, ".csv")}`;
+    }
```
Comment (timwekkenbc) [inline app/utils/api.ts]: Could potentially just use "getShortFilenameOnly". Like "filename = `${scenarioName} ${getShortFilenameOnly(filepath)}`"

File: app/components/ScenariosManager/IsolationTester/IsolationTester.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  resetTrigger,
+  jsonFile,
+  rulemap,
+  scenarioName,
+  ruleContent,
+  ruleVersion,
+}: IsolationTesterProps) {
+  const [simulationRun, setSimulationRun] = useState(false);
+  const [scenarioExpectedOutput, setScenarioExpectedOutput] = useState({});
+  const [editingScenario, setEditingScenario] = useState(scenarioName && scenarioName.length > 0 ? true : false);
```
Comment (timwekkenbc) [inline app/components/ScenariosManager/IsolationTester/IsolationTester.tsx]: These aren't really used? I think you can remove them and adjust/remove the useffects below.

**Review by timwekkenbc — APPROVED** [review]
Looks good! Need to remove some dead code from "IsolationTester".

File: app/components/ScenariosManager/IsolationTester/IsolationTester.tsx (hunk 3 above)
Comment (brysonjbest) [reply app/components/ScenariosManager/IsolationTester/IsolationTester.tsx]: Missed when copying over from the Scenarios Generator and Viewer components! Thank you, definitely unused.

**Pull request merged**